std/zig/parse.zig

```diff
@@ -1,3659 +1,3044 @@
 const std = @import("../std.zig");
 const assert = std.debug.assert;
-const mem = std.mem;
+const Allocator = std.mem.Allocator;
 const ast = std.zig.ast;
-const Tokenizer = std.zig.Tokenizer;
-const Token = std.zig.Token;
+const Node = ast.Node;
+const Tree = ast.Tree;
+const AstError = ast.Error;
 const TokenIndex = ast.TokenIndex;
-const Error = ast.Error;
+const Token = std.zig.Token;
+const TokenIterator = Tree.TokenList.Iterator;
 
-/// Result should be freed with tree.deinit() when there are
-/// no more references to any of the tokens or nodes.
-pub fn parse(allocator: *mem.Allocator, source: []const u8) !*ast.Tree {
+pub const Error = error{UnexpectedToken} || Allocator.Error;
```

Comment:
`UnexpectedToken` is probably the wrong name for this error. Probably `ParseError` instead. (This error is never returned from `pub fn parse`.)